src/System.Runtime.Extensions/src/System/AppDomain.cs

```diff
@@ -26,5 +27,147 @@ public partial class AppDomain
                 AppContext.UnhandledException -= value;  
             }  
         }  
+
+        public string DynamicDirectory => null;
+        public string FriendlyName => "DefaultDomain";
+        public int Id => DefaultADID;
+        public bool IsFullyTrusted => true;
+        public bool IsHomogenous => true;
+        public bool ShadowCopyFiles => false;
+        public event System.EventHandler DomainUnload
+        {
+            add  
+            {  
+                throw new PlatformNotSupportedException();
```

Comment:
Ah, I [see](https://msdn.microsoft.com/en-us/library/system.appdomain.domainunload(v=vs.110).aspx) that this event is never raised for the default domain on desktop. But on desktop you can still add/remove without it throwing, right? Should this do the same instead of throwing?